### DISCUSSION TITLE
Detect menubar popup open

### DISCUSSION THREAD

**Discussion by scha384** [mynameiskenlee/flutter_macos_menubar_example#7] [General]
I've started making a menu bar app with this example by following the steps described.

Since I need a page reload when the app appears, I've already added `focus_detector` for that purpose. I am now looking for a way to detect when the popup is opened.

This needs to be without breaking compatibility with the other platforms that don't use a menu bar (e.g. iOS, Android).

**Comment by mynameiskenlee** [comment]
The popup is handled by showPopover and hidePopover function at [macos/Runner/StatusBarController.swift](https://github.com/mynameiskenlee/flutter_macos_menubar_example/blob/b5bbfa804c8029ce011e5914a68032e6ed716103/macos/Runner/StatusBarController.swift).

You may take a look at [Flutter Platform Channels](https://docs.flutter.dev/platform-integration/platform-channels) and it should be able to detect the popup without breaking compatibility with the other platforms

**Comment by scha384** [comment]
Thanks for the quick reply, that was the solution.

_Some helpful articles for anyone looking at platform channels in the future:
[stackoverflow.com/questions/56201503](https://stackoverflow.com/questions/56201503#56201560)
[stackoverflow.com/questions/50187680](https://stackoverflow.com/questions/50187680)_
